doc/workflows.rst

```diff
@@ -0,0 +1,505 @@
+  # behaviour similar to: bisector run ... 2>&1 | tee myreport.yml.gz.log
+  bisector run --steps steps.yml --report myreport.yml.gz
+
+All available step classes along with available ``run`` options can be looked
+up using ``bisector step-help``. Options are documented in their CLI form, but
+also equally apply to the steps configuration file.
+
+.. tip:: Bisector supports executing commands in a transient systemd scope
+  using ``systemd-run``. This ensures that all processes started indirectly by
+  the command will be terminated/killed when the step finishes, just like for a
+  systemd service. That is a good idea to enable for long-running sessions.
+
+Analyzing results
++++++++++++++++++
+
+
+Reports generated using ``bisector run`` can be inspected using ``bisector
+report``. The expected output of the example configuration could look like
+that if everything went well:
+
+::
+
+  flash/flash (flash) [GOOD]
+      command: <your reboot command>
+      #1 : OK
+      #2 : OK
+      #3 : OK
+      #4 : OK
+      #5 : OK
+
+  boot/reboot (reboot) [GOOD]
+      command: <your reboot command>
+      #1 : OK
+      #2 : OK
+      #3 : OK
+      #4 : OK
+      #5 : OK
+
+  shell/shell (shell) [GOOD]
+      command: sshpass -p password ssh-copy-id -i $HOME/.ssh/id_rsa "$USER@$HOSTNAME"
+      #1 : OK
+      #2 : OK
+      #3 : OK
+      #4 : OK
+      #5 : OK
+
+  test/behaviour (exekall-LISA-test) [BAD]
+      OneSmallTask[board=juno-r0]:test_slack:                      passed 163/163 (100.0%)
+      OneSmallTask[board=juno-r0]:test_task_placement:             passed 163/163 (100.0%)
+      Error: 0/2, Failed: 0/2, Undecided: 0/2, Skipped: 0/2, Passed: 2/2
+
+  my-other-test/test (test) [GOOD]
+      command: hello world
+      #1 : OK
+      #2 : OK
+      #3 : OK
+      #4 : OK
+      #5 : OK
+
+  Overall bisect result: good commit
+
+There is one section per step, reflecting the steps configuration. Each step
+will aggregate the results of all its iterations. The header is formatted as
+*<step name>/<step category> (step class name) [<step result>]*. The overall
+bisect result is the combination of the result of each steps.
+
+``exekall-LISA-test`` has special support for inspecting ``exekall``'s database
+collected during each iteration of ``bisector``, and can display a summary
+table. By default, a *passed* label will only appear if all iteration
+successfully passed.  Otherwise, an appropriate combination of *FAILED*,
+*ERROR*, *SKIPPED* and *UNDECIDED* lines will be displayed with the
+corresponding count.
+
+Various options can affect what is displayed and taken into account. For
+example, ``--skip my-other-test`` will remove the contribution of that step to
+the final result. Step-specific report options are documented in ``bisector
+step-help``. Some of the options allow exporting collected artifacts from the
+report, like ``-o export-logs``. In the case of ``exekall-LISA-test`` step,
```

Comment:
Über-nit: We use the ``-o <option>`` format here, but further down we use ``-o<option>``.

Reply:
Actually both are allowed, since it is a regular gnu-style one-letter-option.

Reply:
Yeah I know, which is why it's an über-nit - it's just to keep some form of consistency :)